pkg/stub/handler.go

```diff
@@ -751,51 +773,61 @@ func (h *Handler) CleanUpOpenshiftNamespaceOnDelete(srcfg *v1alpha1.SamplesResou
 	iopts := metav1.ListOptions{LabelSelector: v1alpha1.SamplesManagedLabel + "=true"}
 
 	streamList, err := h.imageclientwrapper.List("openshift", iopts)
-	if err != nil {
+	if err != nil && !kerrors.IsNotFound(err) {
 		logrus.Warnf("Problem listing openshift imagestreams on SamplesResource delete: %#v", err)
+		return err
 	} else {
-		for _, stream := range streamList.Items {
-			if _, ok := h.skippedImagestreams[stream.Name]; ok {
-				continue
-			}
-			err = h.imageclientwrapper.Delete("openshift", stream.Name, &metav1.DeleteOptions{})
-			if err != nil && !kerrors.IsNotFound(err) {
-				logrus.Warnf("Problem deleteing openshift imagestream %s on SamplesResource delete: %#v", stream.Name, err)
+		if streamList.Items != nil {
+			for _, stream := range streamList.Items {
```

Comment:
we can't just delete every imagestream that's not in our skippedlist.  Users(admins) could have created additional imagestreams in the openshift namespace.  We need to check an annotation or something to confirm it was created/is-managed by us.